.dotnet/api/OpenAI.netstandard2.0.cs

```diff
@@ -1146,14 +1146,24 @@ public class BatchClient {
         public BatchClient(ApiKeyCredential credential, OpenAIClientOptions options = null);
         protected internal BatchClient(ClientPipeline pipeline, Uri endpoint, OpenAIClientOptions options);
         public virtual ClientPipeline Pipeline { get; }
+        public virtual BatchOperation CreateBatch(ReturnWhen returnWhen, BinaryContent content, RequestOptions options = null);
+        public virtual Task<BatchOperation> CreateBatchAsync(ReturnWhen returnWhen, BinaryContent content, RequestOptions options = null);
+        public virtual ClientResult GetBatches(string after, int? limit, RequestOptions options);
+        public virtual Task<ClientResult> GetBatchesAsync(string after, int? limit, RequestOptions options);
+    }
+    public class BatchOperation : OperationResult {
+        public override bool IsCompleted { get; protected set; }
+        public override ContinuationToken? RehydrationToken { get; protected set; }
         public virtual ClientResult CancelBatch(string batchId, RequestOptions options);
         public virtual Task<ClientResult> CancelBatchAsync(string batchId, RequestOptions options);
-        public virtual ClientResult CreateBatch(BinaryContent content, RequestOptions options = null);
-        public virtual Task<ClientResult> CreateBatchAsync(BinaryContent content, RequestOptions options = null);
         public virtual ClientResult GetBatch(string batchId, RequestOptions options);
         public virtual Task<ClientResult> GetBatchAsync(string batchId, RequestOptions options);
-        public virtual ClientResult GetBatches(string after, int? limit, RequestOptions options);
-        public virtual Task<ClientResult> GetBatchesAsync(string after, int? limit, RequestOptions options);
+        public static BatchOperation Rehydrate(BatchClient client, ContinuationToken rehydrationToken, CancellationToken cancellationToken = default);
```

Comment:
It calls a service method that takes a cancellationToken -- it felt like we should let the end-user provide this in case they wanted to cancel the request.

Reply:
why does it call the service?

Reply:
To populate the Value property and find out if the operation has completed?  Should it not?  Maybe I have misunderstood the semantics of the Rehydrate operation.

Reply:
It does look like the Azure.Core-based Rehydrate implementation also makes a service call, it just passes `default` for the CancellationToken where it is doing that: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Operation.cs#L37